pkg/image/validation/kubernetes.go

```diff
@@ -117,6 +113,48 @@ func validateNodes(k8s *image.Kubernetes) []FailedValidation {
 	return failures
 }
 
+func validateNetwork(k8s *image.Kubernetes) []FailedValidation {
+	var failures []FailedValidation
+	ip := k8s.Network.APIVIP
+
+	numNodes := len(k8s.Nodes)
+	if numNodes <= 1 {
+		// Single node cluster, node configurations are not required
+		return failures
+	}
+
+	if ip == "" {
+		failures = append(failures, FailedValidation{
```

Comment:
Returning after this would obsolete the check below and reduce code indentation.